### ISSUE TITLE
BSIP77: Require Higher CR When Creating/Adjusting Debt Positions

### ISSUE THREAD

**Issue by abitmore** [bitshares/bsips#161]
BSIP: 0077
    Title: Require Higher CR When Creating/Adjusting Debt Positions
    Authors: Abit More
             @shulthz
    Status: Draft
    Type: Protocol
    Created: 2019-09-30

# Abstract

This BSIP proposes a protocol change which aims to improve overall collaretal ratio (CR) of smartcoins.

# Motivation

Risk-loving traders tend to borrow as much debt as possible with as little collateral as possible,
which impacts overall CR of that smartcoin. Smartcoin owners would like to have a tool to counter
this behavior.

# Rationale

Requiring a higher CR for new debt positions would introduce a buffer thus would increase the
stability of the smartcoin.

# Specification

Add one parameter "Initial Collateral Ratio" (ICR) to each smartcoin asset, which can be updated by the asset owner.

When adjusting a position, apply [BSIP30](https://github.com/bitshares/bsips/blob/master/bsip-0030.md)
if current CR is lower than ICR but not MCR.

Before the protocol upgrade:
```
when creating a new debt position
  require CR > MCR

when adjusting a debt position
  if new CR <= MCR
    require CR increased and debt not increased

when current CR <= MCR
  trigger margin call
```

After the protocol upgrade:
```
Require ICR >= MCR

when creating a new debt position
  require CR > ICR

when adjusting a debt position
  if new CR <= ICR
    require CR increased and debt not increased

when current CR <= MCR
  trigger margin call
```

Note: logic around *target collateral ratio* (TCR) is not affected.

# Discussion

This BSIP introduces a new tool for smartcoin asset owners to fine tune their assets.
Having an ICR greater than MCR would probably result in less risk-loving traders borrowing that
smartcoin, thus may lead to fewer market activities and worse liquidity. It's up to the asset owners
to decide whether to use this tool.

# References
* https://github.com/bitshares/bsips/issues/129#issuecomment-483937118
* https://github.com/bitshares/bsips/issues/96#issuecomment-484126930
* https://bitsharestalk.org/index.php?topic=29638.0

# Copyright
This document is placed in the public domain.

**Comment by clockworkgr** [comment]
Would this make TCR be X at minimum too?

**Comment by abitmore** [comment]
I don't think we need to limit TCR.

**Comment by pmconrad** [comment]
If I understood @bitcrab correctly, the purpose of lowering MCR is to make it easier to create new tokens and thus to increase the supply.
Introducing a new X that limits token creation to a higher-than-MCR collateral ratio will have the opposite effect.

**Comment by abitmore** [comment]
Introducing a new X will keep the asset farther from GS (or GS protection). It may push premium higher in a down trend.

There is the impossible trinity.

**Comment by bangzi1001** [comment]
1. "when creating a new debt position require CR > MCR"
I don't think this is useful because if MCR is 1.75, traders just set it to 1.76 or 1.7500001 if they are able too.

2. "when adjusting a debt position if new CR <= MCR  require CR increased and debt not increased"
I agree this is very useful for traders exit from margin call territory but how this work if their accounts don't have extra BTS and BitAsset?

3.  Global settlement protection is just temporary. Please don't forget to replace global settlement with individual settlement. Force sell those positions that reach MCR = 1. Something like this:

![2019-04-25_21h05_57](https://user-images.githubusercontent.com/36911788/56737888-02253f80-679e-11e9-8954-dd313c63eec3.jpg)

**Comment by abitmore** [comment]
@shulthz please explain how 2 parameters would work, and what's the pros and cons, and why. Thanks.

**Comment by shulthz** [comment]
> 
> 
> @shulthz please explain how 2 parameters would work, and what's the pros and cons, and why. Thanks.

What i think maybe like this:
Maintenance CR=X?
Max Call Ratio=MACR=MCR?
Custom Call Ratio=CCR
Min Call Ratio=MICR

> let X >= MACR
> let MICR<=CCR<=MACR

> when creating a new debt position
>   require CR > X
> 
> when adjusting a debt position
>   if MICR<=new CR <= X
>   require collateral not decrease and debt not increased

> when current CR <= CCR
>   trigger margin call

The PRO:
1. If the price does not fall sharply, there will not be had a large number of margin call in the market;
The present of margin call method will easily cultivate the market buying inertia;
The current DEX seemed 100 times futures leverage, and will be margin call with a slight feed price change.
2. The amount of MPA will be not decreased significantly in the early stages of prices declining;
3. Because the market decline is iterative: falling, rebounding, falling, and rebounding, this will always make the buffer price space for the margin call;
4. This will weake the effect of the feed price on the DEX, and the DEX price will be closer to the CEX price in the most of the time;
5. The debter can chose the risk which they can carry;
6. From 1.75 to 1.6, the price needs to fall more than 8.5%, from 1.6 to 1.35, and the price needs to fall 15.6%.

The cons: 
1. Have the critical point;
2. May be encouraged the aggressive mortgage, but it maybe not be more aggressive than now;

**Comment by abitmore** [comment]
@shulthz apparently you introduced much more parameters which made the mechanism much more complicated. Can you explain them one by one, and set guidelines for adjusting them (e.g. under what conditions who should tune which parameter in which direction and how much offset)?

If people who are responsible for using/adjusting a parameter don't understand it, don't know when should do what, don't know the boundaries, we won't achieve desired results. Tools should be easy to use. BSIP42 was an example, due to lack of understanding as well as guidelines and also skills/interest, witnesses who meant to be responsible for fine tuning the parameters failed to do so.

**Comment by shulthz** [comment]
1. Max Call Ratio and Min Call Ratio will be set by the community/asset issuer.
2. CCR will be set by the debter, 
The default CCR will be Max Max Call Ratio, if the debter didn't set it.
3. Max Call Ratio and Min Call Ratio was the limits of the CCR.

![图片](https://user-images.githubusercontent.com/34892308/56864507-d3a1a180-69f5-11e9-8b74-af97ea23ff58.png)

**Comment by bitcrab** [comment]
frankly speaking, I don't think "a new debt position could be put into margin call territory very soon" is a problem, we need to accept that there are always margin call orders come out or stayed there.

**Comment by shulthz** [comment]
> 
> 
> we need to accept that there are always margin call orders come out or stayed there.

Frankly speaking, we don't need it, it's a cancer.

**Comment by froooze** [comment]
To make things more simple we need 3 parameter:
- MCR = 1.75
- CRC (margin call) = 1.6
- TCR >= CRC

People can only borrow till MCR, but get only a margin called when CR < CRC. 
This prevents bad debt on the one side, but doesn't punish the borrower when prices dropes only little or suppress prices or removes bitAsset liquidity.

**Comment by abitmore** [comment]
To be clear, "what tools to provide" and "how to use the tools" are different topics. The new parameters and new mechanisms are the "tools" we're discussing. Even the committee won't use any of the new tools on the committee-owned MPAs E.G. bitCNY or etc, IMHO the new tools would bring value for private MPA owners, thus would bring value for the whole ecosystem. So while the community/committee is discussing what's the best value after we have the new parameters, we can get the new parameters added to the system first, the bottom line is "by default, current behavior won't change".

In OP I didn't mention TCR , because it doesn't decide *when* to trigger margin calls. However, it's good to clarify the behavior of TCR when we have the new CR parameter. IMHO, we don't change the behavior: if the TCR set by a user is less than MCR, the "effective TCR" is MCR, where MCR is the threshold to trigger margin calls as defined in OP.

* Optionally we can have a new "minimum_effective_tcr" parameter that can be defined by asset owners, which can be higher than MCR, so when a debt position is being margin called, it sells more at one time, to leave some room for the price to drop without triggering a new margin call.

Other related topics:
* "User-defined call CR" aka "manual margin call" is being discussed as BSIP62 in https://github.com/bitshares/bsips/issues/156 and also https://github.com/bitshares/bsips/issues/129;
* global-settlement protection related topics are being discussed in https://github.com/bitshares/bsips/issues/135, currently there is BSIP58 to define how feed producers should act, technically we can add an on-chain parameter that can be decided by asset owners, which needs a new BSIP;
  * also there are discussions about using a fund to handle bad debt positions, and/or charging additional fees when margin call occurs: https://github.com/bitshares/bsips/issues/164
* "Who should control MCR and MSSR, price feed producers or asset owners"  is being discussed in https://github.com/bitshares/bsips/issues/96 and BSIP59;
* force-settlement related: https://github.com/bitshares/bsips/issues/160

Lots of topics. We should try to implement them all in one consensus release (IMHO best implement them as configurable parameters when possible so current behavior do not change).

**Comment by shulthz** [comment]
It's time to choose.

**Comment by froooze** [comment]
> It's time to choose.

Setting MCR = 1.55 but only do margin calls on positions, which have a lower CR than 1.4.

**Comment by shulthz** [comment]
> 
> 
> > It's time to choose.
> 
> Setting MCR = 1.55, but only do margin calls on positions, which have CR < 1.45.

I think it is not very safe, maybe Maintenance CR=1.6, Margin Call Ratio=1.5

**Comment by ryanRfox** [comment]
This discussion requires someone to draft a formal BSIP. Do I have any volunteers?

**Comment by abitmore** [comment]
@ryanRfox please assign a number for this BSIP. I'll draft one.

**Comment by ryanRfox** [comment]
Assigned BSIP77 @abitmore

**Comment by abitmore** [comment]
Thank you @ryanRfox. Created PR https://github.com/bitshares/bsips/pull/226.

**Comment by froooze** [comment]
@abitmore 
Your BSIP was not clear with X as description. So you use `MCR` as call ratio and `X` as amount to borrow from blockchain?

**Comment by abitmore** [comment]
@froooze OP updated, also see #226.

**Comment by shulthz** [comment]
I think we should pull BSIP74 and BSIP 77.

**Comment by sschiessl-bcp** [comment]
Is that BSIP considered completed @abitmore and all concerns from reviews adressed?

**Comment by abitmore** [comment]
@sschiessl-bcp there is no comment in the PR (#226). Personally I do think it's completed.

**Comment by shulthz** [comment]
I think this BSIP can cooperate with #160。

**Comment by sschiessl-bcp** [comment]
```
when adjusting a debt position
  if new CR <= ICR
    require CR increased and debt not increased
```
Can I really not increase the debt if my CR ends up > ICR?

**Comment by abitmore** [comment]
If your CR ends up > ICR/MCR, it means your "new CR > ICR", so you're not limited by the requirements under the condition "if new CR <= ICR".

**Comment by litepresence** [comment]
"The initial margin requirement is the margin required when purchasing securities, which currently must be at least 50%. The maintenance margin is the amount of equity that must be maintained in the margin account going forward. The minimum maintenance margin requirement set by Reg T is 25%."
https://www.investopedia.com/terms/i/initialmargin.asp

These are the standards set forth by FINRA and US Federal Reserve for margin trading in FOREX. 

In terms spoken of here that would be 1.50 and 1.25 respectively.    Certainly a good move to head towards towards that direction.   So, I fully support an ICR greater than the MCR.    

However "normal" would never work in this environment, nor fix any of the actual issues at hand...  because:

MSSR on the other hand is usually a proprietary number which is enacted by the exchange in order to maintain profitability.   Unlike the 102 MSSR used by Bitshares - enacted humorously by those in debt, rather than those responsible for the chain - in industry, MSSR would be 200 to 10000; ie when you lack collateral liquidation is immediate to between 50 cents and a penny on the dollar.  That's the way margin calls are supposed to work.  That is how exchanges stay liquid; through liquidation of mal investment.  

The system we have here is broken.  MCR and ICR have little to do with it because your MCR is already well above "normal" "nanny state" regulations.   The issue at hand - the reason bitassets are broken - is the absurdly low MSSR, coupled with statutorily fraudulent BSIP76 price feeds.   

While enacting an ICR is a "nice thing to do" for novice margin holders.... Don't lose sight of what is the correct thing to do for the exchange itself.   Double... triple... maybe 5 or 10X the MSSR.   

Liquidation is the only thing that can purge rottenness out of the system.

**Comment by abitmore** [comment]
@litepresence about MSSR, here is my reply. Short version: don't treat bitasset markets as "normal" margin trading markets, look deeper.

Longer version: Not like other margin trading systems where a) there are usually sufficient external supply (aka not yet on the order book) on both side and b) if unable to close an undercollateralized position due to liquidity issue the exchange can take legal actions against the trader thereafter, bitasset (supply) is **created** by margin trading in BitShares, that being said, the main assets that are being priced in the system are bitassets, but not the collateral (BTS). In a traditional market, E.G. BTC/USD market (trading BTC with USD as collateral), there are theoretically unlimited supply of USD but limited supply of BTC, so when money comes, BTC price raises, when money leaves, BTC price falls. In the bitasset markets (trading bitassets with BTS as collateral), when money comes, bitasset price raises, when money leaves, bitasset price falls, regardless of what price BTS is in other markets (e.g. USD). Now the price feeding + margin call + force settlement mechanism is trying to manipulate the market price of bitassets to archieve "the peg", this is why all the strange things come out: force-settlement - you can forcefully buy unlimited amount at a fixed price, MSSR - you're not liquidated when nobody wants to buy at a high-enough price, global settlement - others' well-collateralized positions will be closed when your position is undercollateralized, etc, and later we came with topics about adjusting MCR/MSSR (which led to BSIP42), global-settlement protection, BSIP76, and etc. However, due to the [Impossible Trinity](https://en.wikipedia.org/wiki/Impossible_trinity), it's impossible to maintain the pegged price under all circumstances (see [link](https://bitsharestalk.org/index.php?topic=11946.0) and [link](https://bitsharestalk.org/index.php?topic=27203.0)). It would break sooner or later. Then all the means about rules/parameters adjustment are only to tune "when/where to break", and which direction and how big the price difference is when it breaks (a positive or negative premium) are  acceptable. There is no perfect solution.

**Comment by litepresence** [comment]
I suppose I see things differently.  We should not aim for sovereign monetary policy.   Nor should we aim for a constant peg in the moment.  What we should aim for is honest price feeds and immediate liquidation of under collateralized positions.   We should aim for free markets.   This allowance for price discovery will in the long run always push price back to the mean.   Perhaps not on every trade... You might get some deep spikes... but the TREND will be towards the mean rather than a trend we see today... pushed by BSIP76 and BSIP41; later BSIP42... which push us toward thinner bid/ask margins; thinner high/low... and of course less liquidation; restrained discovery.   But as a consequence we pull further and further from the peg as a result; diverging from the mean... then shocking back towards it upon correction when new policies are implemented.  

Regarding the impossible trinity... monetary sovereignty should be seen as the evil not the aim.  It should not be the state's role to enact monetary policy with aims to "correct" markets.   There is a knowledge problem.  It is only through price DISCOVERY in a free market that inefficiencies can be corrected and brought back to long running pegged mean.   

If the common free market price for bananas is $0.50/lb.  It should not surprise you when one business wishes to close out its over ripe banana stock one day for for a penny a pound... and on the following day another business sells bananas at $5/lb at a hungry music festival full of trust fund kids.   In the end though... we trend back to $0.50.   When the state comes in... they say, "no price gouging!  no flooding the market with cheap product!",  the end result will always be rotten fruit, dissatisfied consumers, and failed businesses. 

http://rationalargumentator.com/issue43/austrianmarketprocess.html

**Comment by abitmore** [comment]
![](https://upload.wikimedia.org/wikipedia/commons/thumb/d/d8/Impossible_trinity_diagram.svg/525px-Impossible_trinity_diagram.svg.png)

@litepresence please be aware that
* we **want** that bitassets be pegged, aka fixed exchange rates, and
* we can't limit free capital flow, and
* the margin call mechanism and force-settlement and price feeding and the related parameters e.g. MCR and MSSR **ARE** monetary policies. No matter MSSR = 1.1 or 110, it is part of the policy.

**Comment by abitmore** [comment]
Draft merged in #226. Closing.
